test/units/modules/cloud/docker/test_docker_swarm_service.py

```diff
@@ -0,0 +1,35 @@
+import pytest
+from units.compat.mock import MagicMock
+
+from ansible.modules.cloud.docker.docker_swarm_service import DockerServiceManager
+from docker.errors import APIError
```

Comment:
Ideally the tests would mock the docker import instead of importing the real module. See an example here:

https://github.com/ansible/ansible/pull/50850/files#diff-d075108943a235e1989fb4a5ac471049R38

If that approach is not being used then any imports outside the Python standard library need to be done such that the tests will be skipped if the imports are unavailable.

See https://docs.pytest.org/en/latest/skipping.html#skipping-on-a-missing-import-dependency for handling imports within the tests themselves.

Reply:
Makes sense to not depend on third part modules when not needed. I've mocked out the docker dependency.